src/eko/evolution_operator/physical.py

```diff
@@ -47,96 +47,75 @@ def ad_to_evol_map(cls, op_members, nf, q2_final, intrinsic_range):
+                n = f**2 - 1
+                m[f"V{n}.V{n}"] = op_members[(br.non_singlet_pids_map["ns-"], 0)]
+                m[f"T{n}.T{n}"] = op_members[(br.non_singlet_pids_map["ns+"], 0)]
+            # deal with intrinsic heavy quark PDFs
+            if intrinsic_range is not None:
+                hqfl = "cbt"
+                op_id = member.OpMember.id_like(
+                    op_members[(br.non_singlet_pids_map["nsV"], 0)]
+                )
+                for intr_fl in intrinsic_range:
+                    if intr_fl <= nf:  # light quarks are not intrinsic
+                        continue
+                    hq = hqfl[intr_fl - 4]  # find name
+                    # intrinsic means no evolution, i.e. they are evolving with the identity
+                    m[f"{hq}+.{hq}+"] = op_id.copy()
+                    m[f"{hq}-.{hq}-"] = op_id.copy()
+        else:
+            m = {
+                "g.g": op_members[(21, 21)],
+                "g.ph": op_members[(21, 22)],
+                "g.S": op_members[(21, 100)],
+                "g.Sdelta": op_members[(21, 101)],
+                "ph.g": op_members[(22, 21)],
+                "ph.ph": op_members[(22, 22)],
+                "ph.S": op_members[(22, 100)],
+                "ph.Sdelta": op_members[(22, 101)],
+                "S.g": op_members[(100, 21)],
+                "S.ph": op_members[(100, 22)],
+                "S.S": op_members[(100, 100)],
+                "S.Sdelta": op_members[(100, 101)],
+                "V.V": op_members[(10200, 10200)],
+                "V.Vdelta": op_members[(10200, 10204)],
+                "Vdelta.V": op_members[(10204, 10200)],
+                "Vdelta.Vdelta": op_members[(10204, 10204)],
+            }
+            # add elements which are already active
+            if nf >= 3:
+                m["Td3.Td3"] = op_members[(br.non_singlet_pids_map["ns+d"], 0)]
+                m["Vd3.Vd3"] = op_members[(br.non_singlet_pids_map["ns-d"], 0)]
+            if nf >= 4:
+                m["Tu3.Tu3"] = op_members[(br.non_singlet_pids_map["ns+u"], 0)]
+                m["Vu3.Vu3"] = op_members[(br.non_singlet_pids_map["ns-u"], 0)]
+            if nf >= 5:
+                m["Td8.Td8"] = op_members[(br.non_singlet_pids_map["ns+d"], 0)]
+                m["Vd8.Vd8"] = op_members[(br.non_singlet_pids_map["ns-d"], 0)]
+            if nf >= 6:
+                m["Tu8.Tu8"] = op_members[(br.non_singlet_pids_map["ns+u"], 0)]
+                m["Vu8.Vu8"] = op_members[(br.non_singlet_pids_map["ns-u"], 0)]
         # deal with intrinsic heavy quark PDFs
         if intrinsic_range is not None:
             hqfl = "cbt"
-            op_id = member.OpMember.id_like(
-                op_members[
-                    (br.non_singlet_pids_map["nsV"], 0)
-                ]  # TODO : Probably it is wrong
-            )
+            if not is_qed:
+                op_id = member.OpMember.id_like(
+                    op_members[(br.non_singlet_pids_map["nsV"], 0)]
```

Comment:
just use, e.g., the `g.g` which is always there

Reply:
the idea was to drop the if condition :upside_down_face: - since, as said, it is always there and here you need just a reference operator to get the size